### PR TITLE
Convert our docs from Sphinx+reST => Mkdocs+md

### PR DESCRIPTION
**Pull request by sehqlr**
Although reST is an awesome documentation format, the Sphinx-specific directives our docs contain don't play nicely with non-Sphinx rendering systems, and not everyone uses reST. Luckily for us, readthedocs supports mkdocs, which works specifically with markdown. 

The code that I used for the conversion is also included in this PR, `preprocessor.py`. I had to preprocess the files for Sphinx directives and turn them into something that pandoc wouldn't ignore. If there need to be changes to the preprocessor, those should be added first, before we reformat any of the `.md` files

What still needs to happen:
- [ ] Make any changes to `preprocessor.py` if needed
- [ ] Delete the old files
- [ ] Write a `mkdocs.yml` file to configure the new backend
- [ ] Decide on a new schema/layout for the roles READMEs. markdown can't include files like reST can

Naturally, any feedback is welcome

### CODE REVIEW

**Comment by siddharthist** [comment]
Eventually, this PR should also delete all the old docs.

**Comment by sehqlr** [comment]
Adding that to the list

**Comment by siddharthist** [comment]
This should also include #1620 and #1626 before it's merged.

File: docs/addons.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+\[../addons/README.rst\](../addons/README.rst)
```
Comment (siddharthist) [inline docs/addons.md]: Looks like these links got escaped, but that shouldn't be a problem since we're redoing our role READMEs.

**Comment by siddharthist** [comment]
I think for the role READMEs, we can literally just 
```
for role in consul docker kubernetes ...; do
  echo "Please see docs/components/${role} for documentation" > "role/${role}/README.md"
done
```
and put them all where they belong, in the docs.

File: docs/components/index.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Components
+
+Mantl is made up of a number of components which can be customized,
+generally using Ansible variables.
+
+\[TOC\] s not visible in the Ansible roles. Here are the links to our
```
Comment (siddharthist) [inline docs/components/index.md]: Looks like TOCs get messed up in the transition.

File: docs/components/index.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+package sources:
+
+>   - [mantl-packaging](https://github.com/asteris-llc/mantl-packaging)
+>   - [mesos-packaging](https://github.com/asteris-llc/mesos-packaging)
```
Comment (siddharthist) [inline docs/components/index.md]: These shouldn't be quoted

File: docs/getting_started/aws.md [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+## Configuring Amazon Web Services for Terraform
+
+Before we can build any servers using Terraform and Ansible, we need to
+configure authentication. We'll be filling in the authentication
+variables for the template located at `terraform/aws.sample.tf`. The
+beginning of it looks like this:
+
+  - \[../../terraform/aws.sample.tf\](../../terraform/aws.sample.tf)
+    
+      - end-before
```
Comment (siddharthist) [inline docs/getting_started/aws.md]: The includes are probably all messed up

File: docs/getting_started/aws.md [hunk 5]
```diff
@@ -0,0 +1,342 @@
+# Amazon Web Services
+
+> **note**
+> 
+> new in version 1.0 multi-az support and terraform modularization
```
Comment (siddharthist) [inline docs/getting_started/aws.md]: We don't need this line

File: docs/getting_started/aws.md [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+
+As of Mantl 0.3 you can bring up Amazon Web Services environments using
```
Comment (siddharthist) [inline docs/getting_started/aws.md]: It's been a looooong time since 0.3

**Comment by siddharthist** [comment]
@sehqlr What's the plan for this PR? Seems like there's a lot here that's not ready.

**Comment by sehqlr** [comment]
I put this one on hold a bit, just because it isn’t as high priority as some of the other things I need to work on. Do you think I should close it and come back to it later when I have more time?

**Comment by siddharthist** [comment]
Let's close it for now, just to keep it out of our queue :-)

**Pull request closed without merging**